src/Gamma/Gamma.jl

```diff
@@ -0,0 +1,56 @@
+using StaticArrays
+import SpecialFunctions: trigamma, digamma, polygamma
+import Distributions: Gamma, shape, rate
+import LogExpFunctions: xlogx
+
+function mean(::ClosedFormExpectation, ::typeof(log), q::Gamma)
+    return digamma(shape(q)) + log(scale(q))
+end
+
+function mean(::ClosedWilliamsProduct, ::typeof(log), q::Gamma)
+    return @SVector [
+        polygamma(1, shape(q)),
+        1/scale(q)
+    ]
+end
+
+function mean(::ClosedFormExpectation, ::typeof(xlogx), q::Gamma)
+    scaler = shape(q)/rate(q)
+    return scaler * (digamma(shape(q)+1) - log(rate(q))) 
+end
+
+function mean(::ClosedFormExpectation, ::typeof(xlog2x), q::Gamma)
+    scaler = shape(q)/rate(q)
+    return scaler * ((digamma(shape(q)+1) - log(rate(q)))^2 + trigamma(shape(q)+1))
+end
+
+function mean(::ClosedFormExpectation, ::ComposedFunction{Square, typeof(log)}, q::Gamma)
+    return trigamma(shape(q)) + (digamma(shape(q)) - log(rate(q)))^2
+end
+
+function mean(::ClosedWilliamsProduct, ::ComposedFunction{Square, typeof(log)}, q::Gamma)
+    return @SVector [
+        polygamma(2, shape(q)) + 2 * (digamma(shape(q)) - log(rate(q)))*trigamma(shape(q)),
+        2 * (digamma(shape(q)) - log(rate(q))) * rate(q)
+    ]
+end
+
+function mean(strategy::ClosedFormExpectation, ::ComposedFunction{Power{Val{3}}, typeof(log)}, q::Gamma)
+    Elogx = mean(strategy, log, q)
+    Elog2x = mean(strategy, Square() ∘ log, q)
+    return polygamma(2, shape(q)) + 3*Elogx * Elog2x - 2*Elogx^3
+end
+
+function mean(strategy::ClosedFormExpectation, p::ComposedFunction{typeof(log), ExpLogSquare{T}}, q::Gamma) where {T}
```

Comment:
Here you using `p` for the second argument name and in the next function you using `f` better to choose only one name for the argument and having a strong justification for changing it.

Reply:
oh, I see you are right. I will change it to `f` then.

Reply:
I have updated the branch, I have changed the notation to make it more consistent.